app/script/components/ephemeralTimer.coffee

```diff
@@ -21,54 +21,32 @@ z.components ?= {}
 
 class z.components.EphemeralTimer
   constructor: (params) ->
-    scheduled_time = new Date(params.expires()).getTime()
-    starting_time = Date.now()
-    timer = scheduled_time - starting_time
-    update_interval = 1000 / 10
+    message_et = params.message
+    ephemeral_expires = new Date(message_et.ephemeral_expires()).getTime()
+    ephemeral_started = new Date(message_et.ephemeral_started()).getTime()
+
+    ephemeral_duration = ephemeral_expires - ephemeral_started
 
     @progress = ko.observable 0
     @remaining_time = ko.observable 0
 
-    @title_caption = ko.pureComputed =>
-      title = ''
-      moment_duration = moment.duration(@remaining_time())
-      if moment_duration.asHours() is 1
-        title += "#{moment_duration.hours()} #{z.localization.Localizer.get_text z.string.ephememal_units_hour}, "
-      else if moment_duration.asHours() > 1
-        title += "#{moment_duration.hours()} #{z.localization.Localizer.get_text z.string.ephememal_units_hours}, "
-
-      if moment_duration.asMinutes() is 1
-        title += "#{moment_duration.minutes()} #{z.localization.Localizer.get_text z.string.ephememal_units_minute} #{z.localization.Localizer.get_text z.string.and} "
-      else if moment_duration.asMinutes() > 1
-        title += "#{moment_duration.minutes()} #{z.localization.Localizer.get_text z.string.ephememal_units_minutes} #{z.localization.Localizer.get_text z.string.and} "
-
-      if moment_duration.asSeconds() is 1
-        title += "#{moment_duration.seconds()} #{z.localization.Localizer.get_text z.string.ephememal_units_second}"
-      else if moment_duration.asSeconds() > 1
-        title += "#{moment_duration.seconds()} #{z.localization.Localizer.get_text z.string.ephememal_units_seconds}"
-
-    @interval_id = window.setInterval =>
-      if Date.now() >= scheduled_time
-        window.clearInterval @interval_id
+    message_et.ephemeral_remaining.subscribe (remaining_time) =>
+      if Date.now() >= ephemeral_expires
+        @progress 1
       else
-        @remaining_time scheduled_time - Date.now()
-        elapsed_time = timer - @remaining_time()
-        @progress elapsed_time / timer
-    , update_interval
+        elapsed_time = ephemeral_duration - remaining_time
+        @progress elapsed_time / ephemeral_duration
```

Comment:
How do I best dispose the interval for the remaining calculation as it is in then message and not the timer.